yarn-project/blob-sink/src/client/http.ts

```diff
@@ -1,33 +1,50 @@
 import { Blob, type BlobJson, computeEthVersionedBlobHash } from '@aztec/blob-lib';
 import { type Logger, createLogger } from '@aztec/foundation/log';
 import { makeBackoff, retry } from '@aztec/foundation/retry';
+import { sleep } from '@aztec/foundation/sleep';
 import { bufferToHex, hexToBuffer } from '@aztec/foundation/string';
 
 import { type RpcBlock, createPublicClient, fallback, http } from 'viem';
 
 import { createBlobArchiveClient } from '../archive/factory.js';
 import type { BlobArchiveClient } from '../archive/interface.js';
 import { outboundTransform } from '../encoding/index.js';
+import type { FileStoreBlobClient } from '../filestore/filestore_blob_client.js';
 import { BlobWithIndex } from '../types/blob_with_index.js';
 import { type BlobSinkConfig, getBlobSinkConfigFromEnv } from './config.js';
-import type { BlobSinkClientInterface } from './interface.js';
+import type { BlobSinkClientInterface, GetBlobSidecarOptions } from './interface.js';
 
 export class HttpBlobSinkClient implements BlobSinkClientInterface {
   protected readonly log: Logger;
   protected readonly config: BlobSinkConfig;
   protected readonly archiveClient: BlobArchiveClient | undefined;
   protected readonly fetch: typeof fetch;
+  protected readonly fileStoreClients: FileStoreBlobClient[];
+  protected readonly fileStoreUploadClient: FileStoreBlobClient | undefined;
 
   constructor(
     config?: BlobSinkConfig,
     private readonly opts: {
       logger?: Logger;
       archiveClient?: BlobArchiveClient;
+      fileStoreClients?: FileStoreBlobClient[];
+      fileStoreUploadClient?: FileStoreBlobClient;
+      /** Callback fired when blobs are successfully fetched from any source */
+      onBlobsFetched?: (blobs: BlobWithIndex[]) => void;
     } = {},
   ) {
     this.config = config ?? getBlobSinkConfigFromEnv();
     this.archiveClient = opts.archiveClient ?? createBlobArchiveClient(this.config);
     this.log = opts.logger ?? createLogger('blob-sink:client');
+    this.fileStoreClients = opts.fileStoreClients ?? [];
+    this.fileStoreUploadClient = opts.fileStoreUploadClient;
+
+    if (this.fileStoreUploadClient && !opts.onBlobsFetched) {
+      this.opts.onBlobsFetched = blobs => {
+        this.uploadBlobsToFileStore(blobs);
+      };
```

Comment:
Nit: avoid mutating arguments if possible. This changes the `opts` object created by the caller.